packages/terra-form-select/src/SingleSelect.jsx

```diff
@@ -170,41 +166,40 @@ class SingleSelect extends React.Component {
 
   render() {
     const {
-      allowClear,
       children,
       defaultValue,
       onChange,
-      placeholder,
       required,
       value,
       intl,
+      isFilterStyle,
       ...otherProps
     } = this.props;
 
-    const defaultPlaceholder = intl.formatMessage({ id: 'Terra.form.select.defaultDisplay' });
-    const selectPlaceholder = placeholder === undefined ? defaultPlaceholder : placeholder;
-    let clearOptionDisplay;
+    let frameValue;
 
-    if (allowClear) {
-      if (selectPlaceholder.length === 0) {
-        clearOptionDisplay = defaultPlaceholder;
+    if (!isFilterStyle || this.state.value || value) {
+      frameValue = SelectUtil.value(this.props, this.state);
+    } else if (children) {
```

Comment:
can this be simplify to :
```js
else {
frameValue = ( children && children[0].props.children ) ? children[0].props.children[0].props.value : children[0].props.value;
```

Reply:
This condition fails if  there is no option/opt-group provided for the select.